broker/src/main/java/io/camunda/zeebe/broker/bootstrap/PartitionManagerStep.java

```diff
@@ -36,72 +36,65 @@ void startupInternal(
             brokerStartupContext.getExporterRepository(),
             brokerStartupContext.getGatewayBrokerTransport());
 
-    CompletableFuture.runAsync(
-        () ->
-            partitionManager
-                .start()
-                .whenComplete(
-                    (ok, error) -> {
-                      if (error != null) {
-                        startupFuture.completeExceptionally(error);
-                        return;
-                      }
+    CompletableFuture.runAsync(partitionManager::start)
```

Comment:
generally this was prone to silently fail if start failed synchronously, thus moving the completion/error handler one layer up.